soccer/src/soccer/WorldState.hpp

```diff
@@ -17,16 +20,52 @@ constexpr double kBallDecayConstant = 0.180;
  * what time it was last seen.
  */
 struct RobotState {
+    using Msg = rj_msgs::msg::RobotState;
+
     Geometry2d::Pose pose;
     Geometry2d::Twist velocity;
     RJ::Time timestamp;
     bool visible = false;
+
+    /**
+     * @brief Default constructor: an invalid robot with visible=false.
+     */
+    RobotState() = default;
+
+    RobotState(const Geometry2d::Pose& pose, const Geometry2d::Twist& velocity,
+               const RJ::Time& timestamp, bool visible)
+        : pose{pose},
+          velocity{velocity},
+          timestamp{timestamp},
+          visible{visible} {}
+
+    /**
+     * @brief Implicit conversion from RobotState::Msg.
+     * @param msg
+     */
+    RobotState(const Msg& msg)
+        : RobotState{msg.pose, msg.velocity, RJ::FromROSTime(msg.stamp),
+                     msg.visible} {}
+
+    /**
+     * @brief Implicit conversion to RobotState::Msg.
+     * @return
+     */
+    [[nodiscard]] operator Msg() const {
+        return rj_msgs::build<Msg>()
+            .stamp(RJ::ToROSTime(timestamp))
+            .pose(pose)
+            .velocity(velocity)
+            .visible(visible);
```

Comment:
We should probably have unit tests to ensure that the `T -> T::Msg -> T` cycle is a nop.

Reply:
added.